src/main/java/com/github/felixgail/GPlayMusicProvider.java

```diff
@@ -218,13 +251,27 @@ public void initialize(@Nonnull InitStateWriter initStateWriter,
                          @Nonnull PlaybackFactoryManager manager) throws InitializationException {
     initStateWriter.state("Initializing...");
     playbackFactory = manager.getFactory(Mp3PlaybackFactory.class);
-    cachedSongs = CacheBuilder.newBuilder()
-        .expireAfterAccess(30, TimeUnit.MINUTES)
-        .initialCapacity(128)
-        .maximumSize(1024)
-        .build(new CacheLoader<String, Song>() {
+    RemovalListener<String, Song> removalListener = new RemovalListener<String, Song>() {
+      @Override
+      public void onRemoval(@Nonnull RemovalNotification<String, Song> removalNotification) {
+        Song song = removalNotification.getValue();
+        try {
+          logFinest("Removing song with id '%s' from cache.", song.getId());
+          Files.deleteIfExists(Paths.get(fileDir.getValue(), song.getId() + ".mp3"));
+        } catch (IOException e) {
+          logWarning(e, "IOException while removing song '%s (%s)'", song.getTitle(), song.getId());
+        }
+      }
+    };
+    CacheBuilder<String, Song> cacheBuilder = CacheBuilder.newBuilder()
+        .expireAfterAccess(Integer.parseInt(cacheTime.getValue()), TimeUnit.MINUTES)
+        .initialCapacity(512)
+        .maximumSize(4096)
```

Comment:
I think that's a bit too much

Reply:
You have to consider, that not only downloaded but every song (20 per search) will be cached

Reply:
Yes, but most of the search results are useless after a few minutes and can be safely removed from the cache. I doubt the least recently used song out of 1000 songs is still needed.